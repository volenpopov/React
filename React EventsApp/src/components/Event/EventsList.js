import React from "react";

import EventItem from "./EventItem";

const EventsList = ({ events, onSetSelectedEventHandler }) => {
    return events.map(event => <EventItem 
        event={event} 
        onSetSelectedEventHandler={onSetSelectedEventHandler}
    />);
};

export default EventsList;