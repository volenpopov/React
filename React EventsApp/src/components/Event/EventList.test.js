import React from 'react';

import { shallow } from 'enzyme';

import EventsList from "./EventsList";

describe("<EventsList />", () => {
    it("expect to match snapshot", () => {
        const mockEvents = [
            {
                id: "1",
                creator: "Pesho",
                date: "2020-09-10T17:00:00.000",
                description: "Test description",
                price: 1,
                title: "Mock Event"
            },
            {
                id: "2",
                creator: "Mitko",
                date: "2020-10-12T17:00:00.000",
                description: "Long text here",
                price: 1,
                title: "Mocker"
            }
        ];

        expect(
            shallow(
                <EventsList
                    events={mockEvents}
                    onSetSelectedEventHandler={() => {}}
                />
            )
        ).toMatchSnapshot();
    });
});
