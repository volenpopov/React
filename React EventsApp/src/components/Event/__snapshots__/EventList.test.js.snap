// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventsList /> expect to match snapshot 1`] = `ShallowWrapper {}`;
